typedmodels/models.py

```diff
@@ -400,7 +400,7 @@ def recast(self, typ=None):
         else:
             raise ValueError("No suitable base class found to recast!")
 
-        if not self.type:
+        if not self.dtm_type:
```

Comment:
Unless django is doing something magic with contenttype foreignkeys (I don't think so) this will do a db query to fetch the ContentType object. When all we really need to check is the local attribute:

```suggestion
        if self.dtm_type_id is None:
```